pkg/pool-hooks/contracts/StableSurgeHook.sol

```diff
@@ -159,12 +147,6 @@ contract StableSurgeHook is BaseHooks, VaultGuard, SingletonAuthentication {
         TokenConfig[] memory,
         LiquidityManagement calldata
     ) public override onlyVault returns (bool) {
-        bool isAllowedFactory = factory == _allowedPoolFactory && IBasePoolFactory(factory).isPoolFromFactory(pool);
```

Comment:
Should we retain the "consistency" check at least?

`IBasePoolFactory(factory).isPoolFromFactory(pool)`

Reply:
I don't think this really adds anything tbh. Do you see any specific case where it matters?

Reply:
It's more of a sanity check when it's factory-deployed. But I guess if we're not deploying from the factory anymore there's no reason to have this.